Xamarin.Essentials/Share/Share.android.cs

```diff
@@ -9,6 +9,12 @@ public static partial class Share
     {
         static Task PlatformRequestAsync(ShareTextRequest request)
         {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+
+            if (string.IsNullOrEmpty(request.Text) && string.IsNullOrEmpty(request.Uri))
+                throw new ArgumentNullException(nameof(request.Text));
+
```

Comment:
Maybe we can move this to the shard layer? Then we don't need to duplicate the code in every platform.